opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -184,27 +188,17 @@ impl crate::trace::Tracer for Tracer {
         };
 
         // Build context for sampling decision
-        let (no_parent, trace_id, parent_span_id, remote_parent, parent_trace_flags) =
-            if let Some(ref parent) = parent_span {
-                let sc = parent.span_context();
-                (
-                    false,
-                    sc.trace_id(),
-                    sc.span_id(),
-                    sc.is_remote(),
-                    sc.trace_flags(),
-                )
-            } else {
-                (
-                    true,
-                    builder
-                        .trace_id
-                        .unwrap_or_else(|| config.id_generator.new_trace_id()),
-                    SpanId::INVALID,
-                    false,
-                    TraceFlags::default(),
-                )
-            };
+        if let Some(sc) = parent_span.as_ref().map(|parent| parent.span_context()) {
```

Comment:
Any reason not to `let trace_id = match parent_span.as_ref() [etc] {};`?

Reply:
Don't have strong feelings either way, seems easier to read if there are either assignments to mut vars or returned data. Mixing both seemed slightly more likely to be misread.